lib/Sema/TypeCheckAttr.cpp

```diff
@@ -737,8 +737,19 @@ void AttributeChecker::visitAccessControlAttr(AccessControlAttr *attr) {
 
   if (auto extension = dyn_cast<ExtensionDecl>(D)) {
     if (attr->getAccess() == AccessLevel::Open) {
-      diagnose(attr->getLocation(), diag::access_control_extension_open)
-        .fixItReplace(attr->getRange(), "public");
+      auto diag =
+          diagnose(attr->getLocation(), diag::access_control_extension_open);
+      diag.fixItRemove(attr->getRange());
+      for (auto Member : extension->getMembers()) {
+        if (auto *FD = dyn_cast<FuncDecl>(Member)) {
+          if (FD->isObjC()) {
+            diag.fixItInsert(FD->getFuncLoc(), "open ");
+          } else {
+            diag.fixItInsert(FD->getFuncLoc(), "public ");
+          }
```

Comment:
We should check if the decl has an access control modifier already; if so, I think we should skip adding the fix-it.

Reply:
Fixed :)